Modules/Quest/DailyQuests.lua

```diff
@@ -1,157 +1,200 @@
 ---@class DailyQuests
-local DailyQuests = QuestieLoader:CreateModule("DailyQuests");
-local _DailyQuests = {}
+local DailyQuests = QuestieLoader:CreateModule("DailyQuests")
+local Questie = _G.Questie
 
 local IsQuestFlaggedCompleted = IsQuestFlaggedCompleted or C_QuestLog.IsQuestFlaggedCompleted
 
 ---@type QuestieMap
-local QuestieMap = QuestieLoader:ImportModule("QuestieMap");
+local QuestieMap = QuestieLoader:ImportModule("QuestieMap")
 ---@type QuestieQuest
-local QuestieQuest = QuestieLoader:ImportModule("QuestieQuest");
+local QuestieQuest = QuestieLoader:ImportModule("QuestieQuest")
 ---@type QuestieTooltips
-local QuestieTooltips = QuestieLoader:ImportModule("QuestieTooltips");
+local QuestieTooltips = QuestieLoader:ImportModule("QuestieTooltips")
 ---@type QuestiePlayer
-local QuestiePlayer = QuestieLoader:ImportModule("QuestiePlayer");
+local QuestiePlayer = QuestieLoader:ImportModule("QuestiePlayer")
 
-local nhcDailyIds, hcDailyIds, cookingDailyIds, fishingDailyIds, pvpDailyIds
+-- Forward declarations
+local nhcDailyIds, hcDailyIds, cookingDailyIds, fishingDailyIds, pvpDailyIds, everydayDailyIds
 
-local lastCheck
 
----@param message string
+local nextCheck = -1 -- -1 is always smaller than GetTime()
+local dailyResetTimer
+
+
+---@param questId number
 ---@return nil
-function DailyQuests:FilterDailies(message, _, _)
-    if message and Questie.db.char.showRepeatableQuests and QuestiePlayer:GetPlayerLevel() == 70 then
-        -- If the REPUTABLE message is empty, i.e contains "::::::::::" we don't count it as a check.
-        if (not lastCheck) and not string.find(message, "::::::::::") then
-            lastCheck = GetTime();
-        elseif lastCheck and GetTime() - lastCheck < 10 and not string.find(message, "::::::::::") then
-            lastCheck = GetTime();
-            return;
+local function _HideDailyQuest(questId)
+    QuestieMap:UnloadQuestFrames(questId)
+    QuestieTooltips:RemoveQuest(questId)
+end
+
+
+---@param questId number
+---@return nil
+local function _ShowDailyQuest(questId)
+    if (not QuestieMap.questIdFrames[questId]) then
+        QuestieQuest:DrawDailyQuest(questId)
+    end
+end
+
+
+--- Reset which daily quests are completed,
+---  can freely be run also at not-daily-reset-time
+--- And Show quests as necessary
+--- Start Timer to reset again
+---@return nil
+local function _ResetEverydayDailyQuests()
+    Questie:Debug(Questie.DEBUG_INFO, "[DailyQuests]: ResetEverydayDailyQuests() - Okey to get run also at non-reset times")
+
+    local dbCharCompletedQuests = Questie.db.char.complete
+
+    for questId, _ in pairs(everydayDailyIds) do
+        if not IsQuestFlaggedCompleted(questId) then
+            -- Mark the quest uncompleted so it can be available again
+            dbCharCompletedQuests[questId] = nil
```

Comment:
Why do we need to do this manually? `Questie.db.char.complete` is filled on QuestieInit by calling the API for all completed quests.

Or does this strange timer code also handles the daily quest reset without a relog? 🤔

Reply:
Yes it handles daily quest reset on the fly without relog or reload or smoothreset. So far only for general (non random) daily quests. (Not yet for random dailies which only one is available per type at once)

Reply:
Guess I have to dig deeper then, to fully understand the timer. Sounds great 👍🏻

Reply:
It is still WIP now as I started to implement more stuff